File: templates/problems/18_bit_manipulation/02_num_one_bits/num_one_bits_test.go
package num_one_bits

import "testing"

func TestHammingWeight(t *testing.T) {
	tests := []struct {
		name     string
		n        uint32
		expected int
	}{
		{"three ones", 0b00000000000000000000000000001011, 3},
		{"one bit", 0b00000000000000000000000010000000, 1},
		{"all ones 32bit", 0b11111111111111111111111111111101, 31},
		{"zero", 0, 0},
		{"power of two", 16, 1},
		{"all ones", 0xFFFFFFFF, 32},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := hammingWeight(tt.n)
			if got != tt.expected {
				t.Errorf("hammingWeight(%d) = %d, want %d", tt.n, got, tt.expected)
			}
		})
	}
}

File: templates/problems/18_bit_manipulation/02_num_one_bits/typescript/numOneBits.test.ts
import { describe, it, expect } from 'vitest';
import { hammingWeight } from './numOneBits';

describe('NumOneBits', () => {
    it('three ones', () => {
        expect(hammingWeight(0b00000000000000000000000000001011)).toBe(3);
    });

    it('one bit', () => {
        expect(hammingWeight(0b00000000000000000000000010000000)).toBe(1);
    });

    it('all ones 32bit', () => {
        expect(hammingWeight(0b11111111111111111111111111111101)).toBe(31);
    });

    it('zero', () => {
        expect(hammingWeight(0)).toBe(0);
    });

    it('power of two', () => {
        expect(hammingWeight(16)).toBe(1);
    });

    it('all ones', () => {
        expect(hammingWeight(0xFFFFFFFF)).toBe(32);
    });
});
